slow_odgi/overlap.py

```diff
@@ -0,0 +1,39 @@
+import sys
+import mygfa
+
+
+
+def getpaths(infile):
+    paths = []
+    for line in mygfa.nonblanks(infile):
+        paths.append(line)
+    return paths
+
+def touches(path1, path2, graph):
+    if path1 == path2:
+        return False
+    for seg_o1 in graph.paths[path1].segments: # (segment, orientation) tuple
+        for seg_o2 in graph.paths[path2].segments:
+            if seg_o1 == seg_o2:
+                return True
+    return False
```

Comment:
I actually don't know if this is any clearer, but we could do this with set intersection:

```py
segs1 = set(graph.paths[path1].segments)
segs2 = set(graph.paths[path2].segments)
return bool(segs1 & segs2)
```